src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -393,16 +398,18 @@ def with_subqueries(self, subquery_plans: List["SnowflakePlan"]) -> "SnowflakePl
             df_aliased_col_name_to_real_col_name=self.df_aliased_col_name_to_real_col_name,
         )
 
-    @cached_property
+    @property
     def attributes(self) -> List[Attribute]:
+        if self._attributes is not None:
```

Comment:
i see we have an output below, which seems just a copy of the attributes, i am really confused about why we have output and attributes? do you know?

Reply:
I think it's from scala snowpark when we try to copy some code at that time.... but I don't know why we have it in scala snowpark.